htdocs/main.php

```diff
@@ -260,30 +268,32 @@ function tplFromRequest($param)
 // This should be array_filter, but to have access to both key and value
 // in array_filter we need to require PHP >= 5.6
 $realPerms = array();
-foreach ($user->getPermissions() as $permName => $hasPerm) {
-    if ($hasPerm === true) {
-        $realPerms[] = $permName;
+if (!$anonymous) {
+    foreach ($user->getPermissions() as $permName => $hasPerm) {
+        if ($hasPerm === true) {
+            $realPerms[] = $permName;
+        }
     }
+    $tpl_data['userPerms']   = $realPerms;
+    $tpl_data['studyParams'] = array(
+                                'useEDC'      => $config->getSetting('useEDC') ?
```

Comment:
Would null coalesce help here?

Reply:
Maybe. I'll let Travis be the judge of that..